cypress/integration/checkin/checkin.basic.spec.js

```diff
@@ -0,0 +1,127 @@
+  const patronDetailsButton = Button('Patron details');
+  const itemDetailsButton = Button('Item details');
+  const newFeeFineButton = Button('New Fee/Fine');
+  const checkInButton = Button('Check in');
+  const availableActionsButton = Button({ id: 'available-actions-button-0' });
+
+  function returnCheckIn() {
+    cy.do(checkInButton.click());
+    cy.do(availableActionsButton.click());
+  }
+
+  before('Create New Item, New User and Check out item', () => {
+    cy.getAdminToken().then(() => {
+      cy.getInstanceTypes({ limit: 1 }).then((instanceTypes) => { itemData.instanceTypeId = instanceTypes[0].id; console.log('itemData.instanceTypeId= ' + itemData.instanceTypeId); });
+      cy.getHoldingTypes({ limit: 1 }).then((res) => { itemData.holdingTypeId = res[0].id; });
+      cy.getLocations({ limit: 1 }).then((res) => { itemData.locationId = res.id; });
+      cy.getLoanTypes({ limit: 1 }).then((res) => { itemData.loanTypeId = res[0].id; });
+      cy.getMaterialTypes({ limit: 1 }).then((res) => { itemData.materialTypeId = res.id; });
+      ServicePoints.getViaApi({ limit: 1 }).then((servicePoints) => { itemData.servicepointId = servicePoints[0].id; });
+    }).then(() => {
+      InventoryInstances.createFolioInstanceViaApi({ instance: {
+        instanceTypeId: itemData.instanceTypeId,
+        title: itemData.instanceTitle,
+      },
+      holdings: [{
+        holdingsTypeId: itemData.holdingTypeId,
+        permanentLocationId: itemData.locationId,
+      }],
+      items:[{
+        barcode: itemData.barcode,
+        status:  { name: 'Available' },
+        permanentLoanType: { id: itemData.loanTypeId },
+        materialType: { id: itemData.materialTypeId },
+      }] });
+    }).then(specialInstanceIds => {
+      itemData.testInstanceIds = specialInstanceIds;
+    });
+
+    cy.createTempUser(userData.permissions)
+      .then(userProperties => {
+        userData.username = userProperties.username;
+        userData.password = userProperties.password;
+        userData.userId = userProperties.userId;
+        userData.barcode = userProperties.barcode;
+        userData.firstName = userProperties.firstName;
+      })
+      .then(() => {
+        UserEdit.addServicePointViaApi(itemData.servicepointId,
+          userData.userId, itemData.servicepointId);
+
+        const payloadInCheckOutItemRequest = {
+          id: uuid(),
+          itemBarcode: itemData.barcode,
+          loanDate: moment.utc().format(),
+          servicePointId: itemData.servicepointId,
+          userBarcode: userData.barcode,
+        };
+
+        checkoutActions.createItemCheckoutApi(payloadInCheckOutItemRequest);
+
+        cy.login(userData.username, userData.password, { path: TopMenu.checkInPath, waiter: CheckInActions.waitLoading });
+      });
+  });
+
+  after('Delete New Service point, Item and User', () => {
+    InventoryInstances.deleteInstanceViaApi(itemData.barcode);
+    cy.get('@userProperties').then(userProperties => {
```

Comment:
you have userId in userData. work through test object instead of values in cy.env

Reply:
I have done changes.